File: app/src/main/java/net/tlalka/android/fiszki/injections/components/ApplicationComponent.java
package net.tlalka.android.fiszki.injections.components;

import dagger.Component;
import net.tlalka.android.fiszki.injections.modules.ApplicationModule;
import net.tlalka.android.fiszki.injections.modules.SessionModule;

import javax.inject.Singleton;

@Singleton
@Component(modules = ApplicationModule.class)
public interface ApplicationComponent {

    SessionComponent add(SessionModule sessionModule);
}

File: app/src/main/kotlin/net/tlalka/fiszki/core/components/ApplicationComponent.kt
package net.tlalka.fiszki.core.components

import dagger.Component
import net.tlalka.fiszki.core.annotations.ApplicationScope
import net.tlalka.fiszki.core.modules.ApplicationModule
import net.tlalka.fiszki.core.modules.SessionModule

@ApplicationScope
@Component(modules = [ApplicationModule::class])
interface ApplicationComponent {

    fun add(sessionModule: SessionModule): SessionComponent
}
